soc/arm/atmel_sam0/common/soc_power.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2019 Benjamin Valentin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <soc_power.h>
+#include <soc.h>
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+static inline void _idle(uint8_t mode) {
+	SCB->SCR &= ~SCB_SCR_SLEEPDEEP_Msk;
+	PM->SLEEP.reg = mode;
+	__DSB();
+	__WFI();
+}
+
+static inline void _sleep(void) {
```

Comment:
Let's call this function `deep_sleep`. Sleep normally means power mode where we don't loose execution context.

Reply:
OK, I named it `sam0_deep_sleep()`